### PR TITLE
Syntevo SmartSynchronize

### PR DESCRIPTION
**Pull request by x80486**

### CODE REVIEW

File: com.syntevo.SmartSynchronize.desktop [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[Desktop Entry]
+Categories=Development;
+Comment=File and Directory Compare
+Encoding=UTF-8
+Exec=smartsynchronize
+Icon=com.syntevo.SmartSynchronize
+Keywords=compare;merge
```
Comment (TingPing) [inline com.syntevo.SmartSynchronize.desktop]: Ends in `;` again

File: com.syntevo.SmartSynchronize.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  "command": "smartsynchronize",
+  "runtime": "org.gnome.Platform",
+  "runtime-version": "3.28",
+  "separate-locales": false,
+  "sdk": "org.gnome.Sdk",
+  "finish-args": [
+    "--device=dri",
+    "--filesystem=host",
+    "--share=ipc",
+    "--socket=wayland",
```
Comment (TingPing) [inline com.syntevo.SmartSynchronize.json]: I doubt this works on Wayland.

File: com.syntevo.SmartSynchronize.json [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        "install -Dm0755 smartsynchronize.sh /app/bin/smartsynchronize",
+        "install -Dm0644 /app/smartsynchronize/bin/smartsynchronize-64.png /app/share/icons/hicolor/64x64/apps/com.syntevo.SmartSynchronize.png",
+        "install -Dm0644 /app/smartsynchronize/bin/smartsynchronize-128.png /app/share/icons/hicolor/128x128/apps/com.syntevo.SmartSynchronize.png",
+        "install -Dm0644 /app/smartsynchronize/bin/smartsynchronize-256.png /app/share/icons/hicolor/256x256/apps/com.syntevo.SmartSynchronize.png",
+        "install -Dm0644 -t /app/share/appdata/ com.syntevo.SmartSynchronize.appdata.xml",
+        "install -Dm0644 -t /app/share/applications/ com.syntevo.SmartSynchronize.desktop"
+      ],
+      "sources": [
+        {
+          "type": "file",
```
Comment (TingPing) [inline com.syntevo.SmartSynchronize.json]: Same comment about `extra-data`.

File: com.syntevo.SmartSynchronize.desktop (hunk 1 above)
Comment (x80486) [reply com.syntevo.SmartSynchronize.desktop]: Since this is a _copycat_ of `SmartGit` I'll address all the comments once the other one is done.

**Comment by nedrichards** [comment]
bot, build com.syntevo.SmartSynchronize

**Comment by x80486** [comment]
Hi @nedrichards, is there any way to know the result for the build?

**Comment by barthalion** [comment]
Results are on https://flathub.org/builds

**Comment by barthalion** [comment]
(For what it's worth, sometimes it also correctly sends status to GitHub, but not too often unfortunately.)

File: com.syntevo.SmartSynchronize.json [hunk 4]
```diff
@@ -0,0 +1,84 @@
+{
+  "app-id": "com.syntevo.SmartSynchronize",
+  "command": "smartsynchronize",
+  "runtime": "org.gnome.Platform",
```
Comment (x80486) [inline com.syntevo.SmartSynchronize.json]: Tried to do the same that I did in `SmartGit` with this one, but I'm getting this beautiful stacktrace at startup:

```shell
Aug 20 21:28:09 uplink com.syntevo.SmartSynchronize.desktop[31734]: Error invoking SmartSynchronize
Aug 20 21:28:09 uplink com.syntevo.SmartSynchronize.desktop[31734]: java.lang.reflect.InvocationTargetException
Aug 20 21:28:09 uplink com.syntevo.SmartSynchronize.desktop[31734]:         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
Aug 20 21:28:09 uplink com.syntevo.SmartSynchronize.desktop[31734]:         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
Aug 20 21:28:09 uplink com.syntevo.SmartSynchronize.desktop[31734]:         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
Aug 20 21:28:09 uplink com.syntevo.SmartSynchronize.desktop[31734]:         at java.base/java.lang.reflect.Method.invoke(Method.java:564)
Aug 20 21:28:09 uplink com.syntevo.SmartSynchronize.desktop[31734]:         at com.syntevo.QBootLoader.main(SourceFile:102)
Aug 20 21:28:09 uplink com.syntevo.SmartSynchronize.desktop[31734]: Caused by: java.lang.UnsatisfiedLinkError: org.eclipse.swt.internal.gtk.OS._cachejvmptr()V
Aug 20 21:28:09 uplink com.syntevo.SmartSynchronize.desktop[31734]:         at org.eclipse.swt.internal.gtk.OS._cachejvmptr(Native Method)
Aug 20 21:28:09 uplink com.syntevo.SmartSynchronize.desktop[31734]:         at org.eclipse.swt.internal.gtk.OS.cachejvmptr(OS.java:486)
Aug 20 21:28:09 uplink com.syntevo.SmartSynchronize.desktop[31734]:         at org.eclipse.swt.internal.gtk.OS.<clinit>(OS.java:97)
Aug 20 21:28:09 uplink com.syntevo.SmartSynchronize.desktop[31734]:         at org.eclipse.swt.internal.Converter.wcsToMbcs(Converter.java:134)
Aug 20 21:28:09 uplink com.syntevo.SmartSynchronize.desktop[31734]:         at org.eclipse.swt.internal.Converter.wcsToMbcs(Converter.java:80)
Aug 20 21:28:09 uplink com.syntevo.SmartSynchronize.desktop[31734]:         at org.eclipse.swt.widgets.Display.<clinit>(Display.java:142)
Aug 20 21:28:09 uplink com.syntevo.SmartSynchronize.desktop[31734]:         at smartsynchronize.hK.a(SourceFile:62)
Aug 20 21:28:09 uplink com.syntevo.SmartSynchronize.desktop[31734]:         at smartsynchronize.xH.a(SourceFile:108)
Aug 20 21:28:09 uplink com.syntevo.SmartSynchronize.desktop[31734]:         at smartsynchronize.KM.a(SourceFile:120)
Aug 20 21:28:09 uplink com.syntevo.SmartSynchronize.desktop[31734]:         at smartsynchronize.yd.a(SourceFile:67)
Aug 20 21:28:09 uplink com.syntevo.SmartSynchronize.desktop[31734]:         at SmartSynchronize.main(SourceFile:11)
Aug 20 21:28:09 uplink com.syntevo.SmartSynchronize.desktop[31734]:         ... 5 more
```

**Comment by nedrichards** [comment]
bot, build com.syntevo.SmartSynchronize

**Comment by nedrichards** [comment]
@x80486 let me know if anyone else from upstream needs to be added to the repo.

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/com.syntevo.SmartSynchronize

**Pull request closed without merging**